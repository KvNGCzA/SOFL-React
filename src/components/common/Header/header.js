import React, { Component } from 'react';
import { Link } from 'react-router-dom';
class Header extends Component {
  render() {
    return (
      <header>
        <h2 className="welcome"><Link to="/">stackOverflowLite</Link></h2>

        <nav className="main-navigation">
          <ul>
            <li><Link to="/">Home</Link></li>
            <li><Link className="profile-link" to="">Profile</Link></li>
            <li><Link to="#" className="logout">Logout</Link></li>
            <li><Link to="/login-signup">Login</Link></li>
          </ul>
        </nav>

        <nav className="mobile-navigation">

          <div className="profile nav-icon">
            <Link className="profile-link" to="">
              <div className="profile-picture">
                <div className="profile-image" />
              </div>
            </Link>
          </div>

          <div className="mob-search nav-icon">
              <i className="fas fa-search" id="fa-search" />
          </div>


          <div className="dropdown-menu nav-icon">
            <i className="fas fa-bars" />
          </div>

        </nav>

        <form className="mob-search-form search">
          <input className="search-form-input" type="text" name="search" placeholder="search" />
          <input type="submit" name="submit" className="" />
        </form>

        <ul className="dropdown-menu-items">
            <li><Link to="/">Home</Link></li>
            <li><Link to="/post-question">Post A Question</Link></li>
            <li><Link to="/login-signup">Login/Sign Up</Link></li>
            <li><Link to="#" className="logout">Log Out</Link></li>
        </ul>
      </header>
    );
  }
}

export default Header;