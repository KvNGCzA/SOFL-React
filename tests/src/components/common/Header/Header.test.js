import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../../../src/components/common/Header/header';


test('NotFound snapshot test', () => {
  const component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});
