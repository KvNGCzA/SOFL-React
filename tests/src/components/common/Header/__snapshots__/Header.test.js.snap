// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound snapshot test 1`] = `
<header>
  <h2
    className="welcome"
  >
    <Link
      replace={false}
      to="/"
    >
      stackOverflowLite
    </Link>
  </h2>
  <nav
    className="main-navigation"
  >
    <ul>
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          className="profile-link"
          replace={false}
          to=""
        >
          Profile
        </Link>
      </li>
      <li>
        <Link
          className="logout"
          replace={false}
          to="#"
        >
          Logout
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/login-signup"
        >
          Login
        </Link>
      </li>
    </ul>
  </nav>
  <nav
    className="mobile-navigation"
  >
    <div
      className="profile nav-icon"
    >
      <Link
        className="profile-link"
        replace={false}
        to=""
      >
        <div
          className="profile-picture"
        >
          <div
            className="profile-image"
          />
        </div>
      </Link>
    </div>
    <div
      className="mob-search nav-icon"
    >
      <i
        className="fas fa-search"
        id="fa-search"
      />
    </div>
    <div
      className="dropdown-menu nav-icon"
    >
      <i
        className="fas fa-bars"
      />
    </div>
  </nav>
  <form
    className="mob-search-form search"
  >
    <input
      className="search-form-input"
      name="search"
      placeholder="search"
      type="text"
    />
    <input
      className=""
      name="submit"
      type="submit"
    />
  </form>
  <ul
    className="dropdown-menu-items"
  >
    <li>
      <Link
        replace={false}
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/post-question"
      >
        Post A Question
      </Link>
    </li>
    <li>
      <Link
        replace={false}
        to="/login-signup"
      >
        Login/Sign Up
      </Link>
    </li>
    <li>
      <Link
        className="logout"
        replace={false}
        to="#"
      >
        Log Out
      </Link>
    </li>
  </ul>
</header>
`;
